client/lib/two-step-authorization/index.js

```diff
@@ -65,6 +67,71 @@ TwoStepAuthorization.prototype.fetch = function( callback ) {
 	);
 };
 
+TwoStepAuthorization.prototype.postLoginRequest = function( endpoint, data ) {
+	if ( ! this.getTwoStepWebauthnNonce() ) {
+		return Promise.reject( 'Invalid nonce' );
+	}
+
+	const url = 'https://wordpress.com/wp-login.php?action=' + endpoint;
+	const formData = new window.FormData();
+	const requestData = {
+		...data,
+		client_id: config( 'wpcom_signup_id' ),
+		client_secret: config( 'wpcom_signup_key' ),
+		auth_type: 'webauthn',
+		two_step_nonce: this.getTwoStepWebauthnNonce(),
+	};
+
+	for ( const key in requestData ) {
+		formData.set( key, requestData[ key ] );
+	}
+
+	return window
+		.fetch( url, {
+			method: 'POST',
+			body: formData,
+			credentials: 'include',
+		} )
+		.then( response => response.json() );
+};
+
+TwoStepAuthorization.prototype.loginUserWithSecurityKey = function( args ) {
+	return this.postLoginRequest( 'webauthn-challenge-endpoint', {
+		user_id: args.user_id,
+	} )
+		.then( response => {
+			const parameters = response.data || [];
+			this.data.two_step_webauthn_nonce = parameters.two_step_nonce;
+			if ( typeof this.data.two_step_webauthn_nonce === 'undefined' ) {
+				return Promise.reject( response );
+			}
+			return webauthn_auth( { publicKey: parameters } );
+		} )
+		.then( assertion => {
+			return this.postLoginRequest( 'webauthn-authentication-endpoint', {
+				user_id: args.user_id,
+				client_data: JSON.stringify( assertion ),
+				create_2fa_cookies_only: 1,
+			} );
+		} )
+		.then( response => {
+			if ( typeof response.success === 'undefined' || ! response.success ) {
+				return Promise.reject( response );
+			}
+			if ( this.isReauthRequired() ) {
```

Comment:
Would be nice if we reuse this piece, it's used in `validateCode` too.

Reply:
Addressed in f7df763efa012a7968abf2947c870524d57f9ad0